app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt

```diff
@@ -183,6 +184,15 @@ class ThreadListAdapter @Inject constructor(
 
     private fun CardviewThreadItemBinding.displayThread(thread: Thread, position: Int) {
 
+        // If we are trying to display an empty Thread, don't. Just delete it.
+        if (thread.messages.isEmpty()) {
+            // TODO: Find why we are sometimes displaying empty Threads, and fix it instead of just deleting them.
+            //  It's possibly because we are out of sync, and the situation will resolve itself shortly? ¯\_(ツ)_/¯
+            threadListAdapterCallback?.deleteThreadInRealm?.invoke(thread.uid)
+            SentryDebug.sendEmptyThread(thread, "No Message in the Thread when displaying it in ThreadList")
+            return
+        }
```

Comment:
Maybe we can move that at the end of the `RefreshController` ? Because otherwise, we can still have `Threads`with 0 message.